### ISSUE TITLE
Cannot spend a Plutus V1 output and pay to a Plutus V2 output with inline datum at the same time

### ISSUE THREAD

**Issue by longngn** [IntersectMBO/cardano-ledger#3180]
CBOR:

```
84a90082825820af0f3e2279948d66514f18627b5615511825c128e26d2a7ae1e1ef66fd0c6add00825820af0f3e2279948d66514f18627b5615511825c128e26d2a7ae1e1ef66fd0c6add01018482583900f65bebbc2bd7920bfdd6d8bc8428adb72a3b4f20d8b5b0c4333689aba1d97c3cedea42ed54c0d89ae7d1b674ed6bbc103f9805e4213c34a01a000ecc1682581d608f514998561d1d3ead2f4c4388c41f2c8f91108b66c4eb501c219825821a001fec62a1581cebe691a343665487dbca5d1853cfb798b68e5ab3e1f699240e2cf999a34568764144411922184568764554481a00559bff4568764d494e1a000b474da300581d70c2b0c5402c47fa30cd62cce1364de781c2d9080f5f1a8109f1c5bd6601821a001ee8e4a2581c90a0b2d0ed365a5cc986f33a5fe08357c7bbe4e2f358744b98eb5c86a148333238353831393401581ce4214b7cce62ac6fbba385d164df48e157eae5863521b4b67ca71d86a15820f3a12ddb59ec26c91d1659ddad054bccd58200046a831327b4d99a7d1c82d4cd190457028201d8185899d8799fd8799fd8799f581cf65bebbc2bd7920bfdd6d8bc8428adb72a3b4f20d8b5b0c4333689abffd8799fd8799fd8799f581ca1d97c3cedea42ed54c0d89ae7d1b674ed6bbc103f9805e4213c34a0ffffffffd8799f581ce4214b7cce62ac6fbba385d164df48e157eae5863521b4b67ca71d865820f3a12ddb59ec26c91d1659ddad054bccd58200046a831327b4d99a7d1c82d4cdff80ff82583900f65bebbc2bd7920bfdd6d8bc8428adb72a3b4f20d8b5b0c4333689aba1d97c3cedea42ed54c0d89ae7d1b674ed6bbc103f9805e4213c34a0821a07ec599aa1581ce4214b7cce62ac6fbba385d164df48e157eae5863521b4b67ca71d86a7582029acf586bf10c3b25c488705a542400178f9c9116f123a581d89fa8fb25ed3fb1a000f012758203bb0079303c57812462dec9de8fb867cef8fd3768de7f12c77f6f0dd80381d0d1a0001e042582044d75b06a5aafc296e094bf3a450734f739449c62183d4e3bbd50c28522afc971a000f5c4c58207339a8bcda85e2c997d9f16beddbeb3ad755f5202f5cfd9cb08db346a1292c011a0007a1205820ea7e9cb359f8b694e6f1ebe593ada33a77f2d7e9881955b1871e5fadaba19df81a000f42405820f3a12ddb59ec26c91d1659ddad054bccd58200046a831327b4d99a7d1c82d4cd1a001069645820f86e870a40ec3cb5f193677ba1aa5c07bf8d3975672557dfcb3a1a2842c1f8411a00101296021a00099941031a00d9e11d0758204fbbf40f710f19b8194b7462ddc4e4e2c29fb84e54deb0a6cb352faee652ca4909a2581c90a0b2d0ed365a5cc986f33a5fe08357c7bbe4e2f358744b98eb5c86a148333238353831393401581cebe691a343665487dbca5d1853cfb798b68e5ab3e1f699240e2cf999a44568764144411922184568764554481a00559bff4568764d494e1a000b474d483332313139303435200b5820513d8f60ae2aec730d7f3ccefa3dd099bdf1c12fe8553feb102903b7cae11f390d8182582012c1ceb0b3f27bdf1e632a4b144804f365854549beb9df36f6536a66b17affd7000e83581cf65bebbc2bd7920bfdd6d8bc8428adb72a3b4f20d8b5b0c4333689ab581c998cb92c7751cddd467e485865fb9374c03f734b2d374938e10436ce581cd620655e34e78d3fda930a673dc8d00201c73b30c4e46e9fab2e37fea50083825820ae34ff787e2a7d7427fd4b9aefa46d2b25179e3c2fcf833723825db065022c5f5840916e38343d9a4846d6674cc654232e391e417f2bfde13ab1d6472441c086c63aaf099c6417f9233239195ca650c77c9b7e4e7b44a8a995f7b6dd73d95b32350182582054d072393f5bdb86e566777c1428731a1e38a4d88288e944ebaea36183bcc44458400f2a0e94a5f71b765c47c2b6479d04363292f59d57221cbbd6f1d57d709aa0b0532241961b5b139ad1a321db80190731ba1f3cc89b1ca3ffd735714266803307825820c69dfb8374d3d556185ede4a86c35c8dcdcd91adf399ffb6655eb951f2f38efa5840202e70fcd759aa678dac1af5d4aa14771d84987b57fc6309cba20eaeef036e62a4ef63818f25a56397c1679552dd6fb57750bf2500dd2301ecb8907553af3a0a01828200581c998cb92c7751cddd467e485865fb9374c03f734b2d374938e10436ce8200581cd620655e34e78d3fda930a673dc8d00201c73b30c4e46e9fab2e37fe03815906f45906f10100003232323232323232323232323232323232323232323232323232323232323232323232232223232323232323553353019003210011622323235301b35301a3500202c01922325335533553335734606800204c2a666ae68c0cc0040540c454cd4ccd54c0b00b4c05804cc0a14040cc08dd71aa80491111110118a99a980419980c3ae5017355335301912233301c22533533018301f002500813300400200110010013302501f35500b034162215335001100222160121300833301875ca02e6a603a6a00a05e02404c04c2a66a666aa605805a602c0266050a02066046eb8d54024888888808c54cd54cd4c064488ccc070894cd4cc060c07c00940204cc0100080044004004cc09407cd5402c0d00548854cd40040a0880644ccd5cd19b87355335300d33301875ca02e6a603a6a00a05e02442a66a0022c4426a00444a66a0062a666ae68c0e40044cc0a80c00085888585888c8cc0cc88cd400520002235002225333573466e3c0080244c01c0044c01800cccc06dd7280d1aa806111111111100398191119a800a4000446a00444a666ae68cdc780100388008980300199b8148001200202902a026026102d1635573a6ea80384c068d4c064d40040ac0614cd4c06c014840045894cd4c014004854cd40040808854cd40040448809007c4c064d40040954ccd5cd18149aab9d0011323301c357420026ae84d5d10009aab9e0010273754006604844a66a0020384426a00444a66a6604400404a260420022600c006604644a66a0020364426a00444a66a66042004048260400022600c006264a66a6a6aa666ae68c090d55ce80089909180080118079aba135573c00204440024466a004034403c420022c6ea8004888008cc048894cd4008010400405c88d400888d400c88c8cd40148cd401094ccd5cd19b8f00200100301b201f2335004201f25333573466e3c00800400c06c54cd400c854cd400884cd40088cd40088cd40088cd40088cc08000800480888cd400880888cc080008004888088888cd401080888894ccd5cd19b8700600315333573466e1c0140084ccd5cd19b87004001023024020020019153350012019007101423301022333500301b0020013500101a13300175ceb488c88c008dd5800980d111999aab9f00120162323301233013300635573a002600a6aae78004c010d5d10019aba1002019233500100c011223018225335001100322133006002300400123004350010022220032325333573460306aae740044c8c8c8c8c848ccc00401400c008d5d09aba2002357420026ae88008d5d08009aab9e0010163754002464a666ae68c05cd55ce80089919191980618031aba1003533357346034002264646020a666ae68c0740044c8c8c8c8c8488ccc00401801000cdd69aba1357440046eb4d5d08009aba2002375a6ae84004d55cf0010a999ab9a301c001130103008357426aae78008068d55ce8009baa357426aae7800854ccd5cd180d80080680c1aab9d00137546ae84d5d10009aab9e001015375400246464a666ae68c0600044c8488c00400cdd71aba135573c0042a666ae68c05c0044c02cdd71aba135573c00402a6aae74004dd5000919191999aa999ab9a3370e90030008990911118020029aba135573c0042a666ae68cdc3a400800226424444600400a6ae84d55cf0010a999ab9a3017001132122223001005357426aae7800854ccd5cd180b0008990911118018029aba135573c004028401c401c401c4666aa60200226601244a66a00442006200201c46464a66aa666ae68cdc79a80100a9a80080a8999ab9a3370e6a00402c6a00202c02002201a2601c00601a600e006600c6a0020266600e464a666ae68c064d55ce800899198061aba1001357426ae88004d55cf00080b9baa00135300635004012222222222200a35573a0026ea8d40040408c94ccd5cd180a1aab9d001132323300853335734602c6aae740044dd71aba135573c0020286eb4d5d09aba200237546ae84004d55cf0008091baa0012325333573460266aae740044c8c8c8c8c8c8c8c8c8c8c8c8c8c8c8c8c8c8c848cccccccccc00404c04403c03402c02401c01400c008d5d09aba2002357420026ae88008d5d08009aba2002357420026ae88008d5d08009aba2002357420026ae88008d5d08009aba2002357420026ae88008d5d08009aba2002357420026aae78004044dd5000919118011bac001300e2233335573e002401446600a60086ae84008c00cd5d1001006910911980080200191091980080180108021091180100191199ab9a3371e004002006008200a200a9111cebe691a343665487dbca5d1853cfb798b68e5ab3e1f699240e2cf99900300422112225335001135003008221333500500930040023335530070080050040011200122002220012323001001223300330020020011533573892010350543100162222222222009370e90001b8748009049fd8799fd8799fd8799f581cf65bebbc2bd7920bfdd6d8bc8428adb72a3b4f20d8b5b0c4333689abffd8799fd8799fd8799f581ca1d97c3cedea42ed54c0d89ae7d1b674ed6bbc103f9805e4213c34a0ffffffffffff0581840000d87a80821a002dc6c01a77359400f5a11902a2a1636d736781774d696e737761703a204661726d204d6967726174696f6e
```

Error when submitting with CLI: `Command failed: transaction submit  Error: Error while submitting tx: ShelleyTxValidationError ShelleyBasedEraBabbage (ApplyTxError [UtxowFailure (UtxoFailure (FromAlonzoUtxoFail (UtxosFailure (CollectErrors [BadTranslation (InlineDatumsNotSupported (TxOutFromOutput (TxIx 2)))]))))])`

**Comment by JaredCorduan** [comment]
hi @longngn :wave: 

What you are seeing is the [expected behavior](https://github.com/input-output-hk/cardano-ledger/blob/064acb8e7ab7b2aeef138ec0c821f8c5f1402621/eras/babbage/test-suite/test/Test/Cardano/Ledger/Babbage/TxInfo.hs#L216-L219).

The `BadTranslation` part of the error is telling us that it could not translate the appropriate ledger state to a plutus script context. The `InlineDatumsNotSupported (TxOutFromOutput (TxIx 2` part of the error is describing that the culprit is the third (index 2) output in the transaction.

Here's the problem: that transaction is running a Plutus V1 script, but the Plutus V1 script context does not allow for inline datums. Remember that in the Alonzo era, there was no such thing as inline datums, that came with the Babbage era and was introduced only for Plutus V2.

When we added inline datums to the ledger, we basically had two choices:
1. Silently ignore parts of the relevant script context when creating a Plutus V1 script context (in this case, a part of a transaction output in the associated transaction).
2. Force a phase 1 validation error.

We chose option 2, since option 1 could lead to some surprising and unfortunate mistakes by DApp developers.

I should write this up as an [ADR](https://github.com/input-output-hk/cardano-ledger/pull/3177)!
